api/api-agents-v6/src/main/java/com/thoughtworks/go/apiv6/agents/AgentsControllerV6.java

```diff
@@ -0,0 +1,255 @@
+import com.thoughtworks.go.server.service.SecurityService;
+import com.thoughtworks.go.server.service.result.HttpLocalizedOperationResult;
+import com.thoughtworks.go.server.service.result.HttpOperationResult;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import com.thoughtworks.go.util.TriState;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Consumer;
+
+import static com.thoughtworks.go.apiv6.agents.representers.AgentRepresenter.toJSON;
+import static com.thoughtworks.go.apiv6.agents.representers.AgentUpdateRequestRepresenter.fromJSON;
+import static com.thoughtworks.go.util.CommaSeparatedString.commaSeparatedStrToList;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toCollection;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static spark.Spark.*;
+
+@SuppressWarnings("ALL")
+@Component
+public class AgentsControllerV6 extends ApiController implements SparkSpringController, CrudController<AgentInstance> {
+    private final AgentService agentService;
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private final SecurityService securityService;
+    private final EnvironmentConfigService environmentConfigService;
+
+    @Autowired
+    public AgentsControllerV6(AgentService agentService, ApiAuthenticationHelper apiAuthenticationHelper,
+                              SecurityService securityService, EnvironmentConfigService environmentConfigService) {
+        super(ApiVersion.v6);
+        this.agentService = agentService;
+        this.apiAuthenticationHelper = apiAuthenticationHelper;
+        this.securityService = securityService;
+        this.environmentConfigService = environmentConfigService;
+    }
+
+    @Override
+    public String controllerBasePath() {
+        return Routes.AgentsAPI.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+            before("", mimeType, this::checkSecurityOr403);
+            before("/*", mimeType, this::checkSecurityOr403);
+
+            get("", mimeType, this::index);
+            get(Routes.AgentsAPI.UUID, mimeType, this::show);
+            patch(Routes.AgentsAPI.UUID, mimeType, this::update);
+            patch("", mimeType, this::bulkUpdate);
+            delete(Routes.AgentsAPI.UUID, mimeType, this::deleteAgent);
+            delete("", mimeType, this::bulkDeleteAgents);
+
+            exception(HttpException.class, this::httpException);
```

Comment:
We don't need to add a HttpException handler at each controller (anymore). 

As part of https://github.com/gocd/gocd/commit/a6d59c3b97e91bf70eb0d4cc278eb802d9d337c2 commit, we have added a top-level handler and removed from individual controllers.

Reply:
Fixed.